app/routes.py

```diff
@@ -1 +1,219 @@
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400))
+        
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
+
+def slack_bot_msg(task_title):
+
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_TOKEN = os.environ.get("SLACK_TOKEN")
+    my_headers = {'Authorization' : 'Bearer '+SLACK_TOKEN}
+    query_params ={
+        "channel": "task-notifications",
+        "text": (f"Someone just completed the task {task_title}.")
+    }
+
+    requests.post(PATH, params=query_params, headers=my_headers)
+
+
+#Task routes
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    sort_param=request.args.get("sort")
+    if sort_param == "asc":
+        tasks=Task.query.order_by(Task.title.asc())
+    elif sort_param == "desc":
+        tasks=Task.query.order_by(Task.title.desc())
+    else:
+        tasks=Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append({
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)
+            }), 200
+    return jsonify(tasks_response),200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+
+    if not task.goal_id:
+        return task.response_dict(), 200
+    else:
+                return jsonify({
+            "task": {
+                "id": task.task_id,
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                }}), 200
+        
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():  
+    request_body = request.get_json()      
+    if ("title" not in request_body) or ("description" not in request_body):
+        return jsonify({
+            "details": "Invalid data"
+        }), 400
+    else:
+        request_body = request.get_json()
+        new_task = Task(title=request_body["title"],
+                description=request_body["description"])
+        db.session.add(new_task)
+        db.session.commit()
+        return new_task.response_dict(), 201
+    
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+    return task.response_dict(), 200
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+    db.session.delete(task)
+    db.session.commit()
+    return make_response(jsonify({
+        "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
```

Comment:
Good job! Another way we can do this to make it a little cleaner is:

```suggestion
        "details": f"Task {task.task_id} '{task.title}' successfully deleted"
```

Using single quotes inside double quotes works.